addons/portal/models/portal_mixin.py

```diff
@@ -76,9 +76,9 @@ def _get_share_url(self, redirect=False, signup_partner=False, pid=None):
     @api.multi
     def _notify_get_groups(self, message, groups):
         access_token = self._portal_ensure_token()
-        customer = self['partner_id']
 
-        if access_token and customer:
+        if access_token and 'partner_id' in self and self['partner_id']:
```

Comment:
```suggestion
        if access_token and 'partner_id' in self._fields:
```
Did it crash with `self['partner_id' == False]` too or just an improvement?

Reply:
No, it doesn't crush, but it's a needed condition. Previously, it was `and customer`, being `customer == self["partner_id"]`, so with my suggestion we were equaling the condition.

Reply:
Oh right, read the diff too fast! Just add the `self._fields` to be explicit and LGTM :+1: